C7x Support for ThreadX

jblanchard-jblopen opened the pull request:
C7x Port, DPL implementation and Examples.

- Added threadx support for the C7x core.
- Added threadx examples for C7x.
- Updated generated files.

shiva-ti commented on source/kernel/threadx/ports/ti_arm_gcc_clang_cortex_c7x/src/tx_threadx_c75.c:
```diff
@@ -0,0 +1,264 @@
+/***************************************************************************
+ * Copyright (c) 2024 Microsoft Corporation
```
@jblanchard-jblopen should we still use Microsoft copyright in newly added code ?

jblanchard-jblopen replied:
We'll review the file to make sure it doesn't contain any previous code and remove the copyright notice if that is the case.

shiva-ti commented on examples/kernel/threadx/hello_world/am62ax-sk/c75ss0-0_threadx/ti-c7000/syscfg_c.rov.xs:
```diff
@@ -0,0 +1,8 @@
+ *  This file contains the information needed by the Runtime Object
+ *  View (ROV) tool.
+ */
+var crovFiles = [
```
@jblanchard-jblopen is this correct ? including FreeRTOS for threadx files

jblanchard-jblopen replied:
That is probably a regenerated file included by mistake in the PR, we'll update the PR.

jblanchard-jblopen committed:
am62a: threadx: C7x Support.

- Added threadx support for the C7x core.
- Added threadx examples for C7x.
- Updated generated files.

Fixes: SITSW-5614
Signed-off-by: Jonathan Blanchard <jonathan.blanchard@jblopen.com>

jblanchard-jblopen commented:
Pull request updated with the following changes:

- Removed generated files specific to FREErtos from the PR.
- Removed spurious copyright header from a new file.
